### PR TITLE
Updates to command parsing

### PR DESCRIPTION
**Pull request by dthigpen**
I have updated the `applyInput` function to handle take, use and direction commands consistently with the rest of the commands.

I also updated the take and use commands to take in multi-word item names, removing ambiguous item resolving when they start with the same prefix.

Let me know if you forsee any issues or have any other ideas in this area!

Thanks

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -758,23 +769,11 @@ let applyInput = (input) => {
...
-    // (just tries to match on the first word)
-    takeItem(args[0]);
-  } else if (command === 'use' && args.length) {
-    // support for using items with spaces in the names
-    // (just tries to match on the first word)
-    useItem(args[0]);
   } else if (args.length >= commands.length) {
     exec(commands[commands.length - 1][command], args);
-  } else if (room.exits && getExit(command, room.exits)) {
-    // handle shorthand direction command, e.g. "EAST" instead of "GO EAST"
```
Comment (okaybenji) [inline index.js:776]: I appreciate what you're trying to do here, but this code currently handles not just cardinal directions but custom exits. For instance, play the demo disk and type `north` followed by `up` to go upstairs.

File: index.js [hunk 2]
```diff
@@ -712,6 +720,9 @@ let commands = [
   // two+ arguments (e.g. "look at key", "talk to mary")
   {
     look: lookAt,
+    take: (args) => takeItem(args.join(' ')),
+    get: (args) => takeItem(args.join(' ')),
+    use: (args) => useItem(args.join(' ')),
```
Comment (okaybenji) [inline index.js:725]: This part looks good, think this is gonna work 👍

**Review by okaybenji — CHANGES_REQUESTED** [review]
Would you please remove the changes related to exits (since they would reduce the funtionality of the engine) and just leave in your improvements to handling of the take/use functions?

Thanks for your PR!

File: index.js (hunk 1 above)
Comment (dthigpen) [reply index.js:776]: Ah yes, that makes sense! That's a nice shorthand. I'll revert this portion

**Comment by dthigpen** [comment]
Reverted the direction command changes, leaving only the take/get/use changes. Let me know if there is anything else!

**Review by okaybenji — APPROVED** [review]

**Comment by okaybenji** [comment]
Perfect, thanks very much!

**Pull request merged**